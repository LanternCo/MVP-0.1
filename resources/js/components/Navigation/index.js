import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { NavLink } from "react-router-dom";
import SignOut from "../Auth/SignOut"
class Navigation extends Component {
    navbarLinks() {
        if (this.props.authenticated) {
            return <SignOut />;
        }

        return [
            <li key="signin" className="nav-item">
                <NavLink to="/signin" className="nav-link">
                    Sign in
                </NavLink>
            </li>,
            <li key="signup" className="nav-item">
                <NavLink to="#" className="nav-link">
                    Sign up
                </NavLink>
            </li>
        ];
    }

    render() {
        return (
            <nav className="navbar navbar-expand-md navbar-light bg-white shadow-sm">
                <div className="container">
                    <a className="navbar-brand" href="/">
                        {this.props.appName}
                    </a>
                    <button
                        className="navbar-toggler"
                        type="button"
                        data-toggle="collapse"
                        data-target="#navbarSupportedContent"
                        aria-controls="navbarSupportedContent"
                        aria-expanded="false"
                        aria-label="{{ __('Toggle navigation') }}"
                    >
                        <span className="navbar-toggler-icon" />
                    </button>

                    <div
                        className="collapse navbar-collapse"
                        id="navbarSupportedContent"
                    >
                        <ul className="navbar-nav ml-auto">
                            {this.navbarLinks()}
                        </ul>
                    </div>
                </div>
            </nav>
        );
    }
}

Navigation.propTypes = {
    appName: PropTypes.string
};

Navigation.defaultProps = {
    appName: "Lantern"
};

function mapStateToProps(state) {
    return {
        authenticated: state.auth.authenticated
    };
}

export default connect(mapStateToProps)(Navigation);
