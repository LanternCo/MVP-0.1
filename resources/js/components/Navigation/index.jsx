import React from 'react';
import PropTypes from 'prop-types';

const Navigation = props => {
    const { appName } = props;

    return (
        <nav className="navbar navbar-expand-md navbar-light bg-white shadow-sm">
            <div className="container">
                {/* @TODO - get app name from props */}
                <a className="navbar-brand" href="/">
                    {appName}
                </a>
                <button
                    className="navbar-toggler"
                    type="button"
                    data-toggle="collapse"
                    data-target="#navbarSupportedContent"
                    aria-controls="navbarSupportedContent"
                    aria-expanded="false"
                    aria-label="{{ __('Toggle navigation') }}"
                >
                    <span className="navbar-toggler-icon" />
                </button>
            </div>
        </nav>
    );
};

Navigation.propTypes = {
    appName: PropTypes.string,
};

Navigation.defaultProps = {
    appName: 'Lantern',
};

export default Navigation;
