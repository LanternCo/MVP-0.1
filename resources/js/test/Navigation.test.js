// Link.react.test.js
import React from "react";
import Navigation from "../components/Navigation";
// @TODO - figure out what the difference between these two are.
import renderer from "react-test-renderer";
import { render, fireEvent } from 'test-utils';

it("Navigation snapshot", () => {
    const component = renderer.create(
        <Navigation appName="Test" />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('Displays appname based on prop', () => {
    const component = renderer.create(
        <Navigation appName="Test" />
    );

    const componentInstance = component.root;
    const componentInstanceProps = componentInstance.props;

    expect(componentInstanceProps.appName).toEqual('Test');
  });
