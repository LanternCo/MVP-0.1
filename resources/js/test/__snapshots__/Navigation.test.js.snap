// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation snapshot 1`] = `
<nav
  className="navbar navbar-expand-md navbar-light bg-white shadow-sm"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      Test
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="{{ __('Toggle navigation') }}"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav ml-auto"
      />
    </div>
  </div>
</nav>
`;
